DependencyInjection/ElaoErrorNotifierExtension.php

```diff
@@ -39,10 +40,19 @@ public function load(array $configs, ContainerBuilder $container)
             $loader = new XmlFileLoader($container, new FileLocator(array(__DIR__ . '/../Resources/config/')));
             $loader->load('services.xml');
 
+            $definition = $container->getDefinition('elao.error_notifier.listener');
+            
             if ($config['mailer'] != 'mailer') {
-                $definition = $container->getDefinition('elao.error_notifier.listener');
                 $definition->replaceArgument(0, new Reference($config['mailer']));
             }
+            
+            if(Kernel::MAJOR_VERSION >= 4) {
```

Comment:
You should not rely on the kernel version but on the new event class existence instead (http-kernel and console are two different components. Checking the kernel version would only give the http-kernel version).

Reply:
Good point, I can just check for the class existence instead.  Will wait to make this change until the other comment on this is responded to, as I don't think we want to listen to both events always.